modules/kubernetes-addons/opentelemetry-operator/main.tf

```diff
@@ -1,10 +1,168 @@
-module "operator" {
-  source        = "../helm-addon"
-  helm_config   = local.helm_config
-  irsa_config   = null
-  addon_context = var.addon_context
+resource "aws_eks_addon" "example" {
+  cluster_name = var.addon_context.eks_cluster_id
+  resolve_conflicts = "OVERWRITE"
+  addon_name   = "adot"
+  depends_on = [kubectl_manifest.opentelemetry-prereq]
+}
 
-  depends_on = [kubernetes_namespace_v1.prometheus]
+resource "kubectl_manifest" "opentelemetry-prereq" {
+    yaml_body = <<YAML
```

Comment:
Aesthetically speaking, I think its best if separate this out into yaml files and read them in via `templatefile()`.

Reply:
This is on me - I asked @saaish to push up what he had so far so I could take a look but I got swamped with other requests. I'll take a look and update